src/AgentBuilder.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+
+namespace Nipwaayoni;
+
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\Psr17FactoryDiscovery;
+use Nipwaayoni\Events\DefaultEventFactory;
+use Nipwaayoni\Events\EventFactoryInterface;
+use Nipwaayoni\Helper\Config;
+use Nipwaayoni\Stores\TransactionsStore;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+class AgentBuilder
```

Comment:
I agree that the `Config` class and its contract need work. I'm not sure the AgentBuilder is the place to do that, though. Eventually, I see all of the `with*Data()` methods being removed from the `AgentBuilder` in favor of strongly typed variants. That's why I made separate methods, so we can more easily deprecate them as we build out better implementations.

How often do you think consumers will benefit from setting individual configuration options? In my experiences, it's easier to construct an array in the context of the creation rather than calling lots of individual methods. Especially if we want the object to remain immutable. Another possible option may be a named constructor in the `Config` class which would hide the internal constructor complexity. Something like:

```
$config = Config::makeFromArray($configData);
...
public static function makeFromArray(array $configData): self
{
  // validate $data array
  return new self($data[...], ...);
}
```
I suspect there are pros and cons to either approach. I'm open to trying either.

Reply:
> How often do you think consumers will benefit from setting individual configuration options? In my experiences, it's easier to construct an array in the context of the creation rather than calling lots of individual methods. Especially if we want the object to remain immutable. Another possible option may be a named constructor in the Config class which would hide the internal constructor complexity. Something like:

You're right. Maybe not so often, but it could help for internal usage. I haven't any stronger argument than this and maybe this no makes sense for now.